src/mint.rs

```diff
@@ -0,0 +1,190 @@
+                Adjustment::IntervalAfterFirstAdjustmentAbsolute(adjust_on_interval, _) => {
+                    adjust_on_interval
+                }
+                Adjustment::IntervalAfterFirstAdjustmentRelative(adjust_on_interval, _) => {
+                    adjust_on_interval
+                }
+            }),
+        }
+    }
+
+    pub fn mint_exact_tokens(&mut self, requested_amount: Balance) -> Result<(), MintingError> {
+        ensure!(
+            self.capacity >= requested_amount,
+            MintingError::NotEnoughCapacity
+        );
+        self.capacity -= requested_amount;
+        self.total_minted += requested_amount;
+        Ok(())
+    }
+
+    pub fn mint_some_tokens(&mut self, requested_amount: Balance) -> Balance {
+        let minted = if self.capacity >= requested_amount {
+            requested_amount
+        } else {
+            self.capacity
+        };
+        self.capacity -= minted;
+        self.total_minted += minted;
+        minted
+    }
+
+    pub fn set_capacity(&mut self, new_capacity: Balance) {
+        self.capacity = new_capacity;
+    }
+
+    pub fn can_mint(&self, amount: Balance) -> bool {
+        self.capacity >= amount
+    }
+
+    pub fn created_at(&self) -> BlockNumber {
+        self.created_at
+    }
+
+    pub fn total_minted(&self) -> Balance {
+        self.total_minted
+    }
+
+    pub fn transfer_capacity_to(
+        &mut self,
+        destination: &mut Self,
+        capacity_to_transfer: Balance,
+    ) -> Result<(), MintingError> {
+        ensure!(
+            self.capacity >= capacity_to_transfer,
+            MintingError::NotEnoughCapacity
+        );
+        self.capacity -= capacity_to_transfer;
+        destination.capacity += capacity_to_transfer;
+        Ok(())
+    }
+
+    pub fn adjustment(&self) -> Option<AdjustOnInterval<Balance, BlockNumber>> {
+        self.adjustment_on_interval
+    }
+
+    /// Updates capacity mints where the adjust_capacity_at_block_number value matches the current block number.
+    /// At such time the value is updated by adding adjustment.block_interval.
+    pub fn maybe_do_capacity_adjustment(&mut self, now: BlockNumber) -> bool {
+        if !self.is_time_for_adjustment(now) {
+            return false;
+        }
+
+        assert!(self.adjustment_on_interval.is_some());
+
+        let adjustment = self.adjustment_on_interval.unwrap();
+
+        // update mint capacity
+        self.adjust_capacity(adjustment.adjustment_type);
+
+        // set blocknumber for next adjustment
+        self.adjust_capacity_at_block_number = Some(now + adjustment.block_interval);
+
+        true
+    }
+
+    fn is_time_for_adjustment(&self, now: BlockNumber) -> bool {
+        self.adjust_capacity_at_block_number
+            .map_or(false, |block_number| block_number == now)
+    }
+
+    fn adjust_capacity(&mut self, adjustment: AdjustCapacityBy<Balance>) {
```

Comment:
Its not safe to make this a method which allows the adjustment of `self` based on an external adjustment policy, despite `self` already having one. You are better of doing

```Rust
self.capacity = adjusted_capacity(self.capacity, adjustment.adjustment_type)
...
fn adjust_capacity(capacity: Balance, adjustment: AdjustCapacityBy<Balance>)
```

Reply:
if adjust_capacity() is private then you couldn't pass in an external adjustment type.  but having adjust_capacity not be a method makes it more testable I suppose.

Reply:
done.